vs_utils/utils/grid_featurizer.py

```diff
@@ -0,0 +1,444 @@
+    2. Generate a second random unit vector thru the algorithm in (1), output v
+      a. If absolute value of u \dot v > 0.99, repeat. This is important for numerical stability 
+        (intuition: we want them to be as linearly independent as possible or else the 
+          orthogonalized version of v will be much shorter in magnitude compared to u. I assume
+          in Stack they took this from Gram-Schmidt orthogonalization?)
+      b. v' = v - (u \dot v)*u, i.e. subtract out the component of v that's in u's direction 
+      c. normalize v' (this isn't in Stack but I assume it must be done)
+    3. find w = u \cross v' 
+    4. u, v', and w will form the columns of a rotation matrix, R. The intuition is that
+        u, v' and w are, respectively, what the standard basis vectors e1, e2, and e3 will be 
+        mapped to under the transformation.
+  '''
+
+  u = generate_random_unit_vector()
+  v = generate_random_unit_vector()
+  while np.abs(np.dot(u,v)) >= 0.99:
+    v = generate_random_unit_vector()
+
+  vp = v - (np.dot(u,v)*u)
+  vp /= np.linalg.norm(vp)
+
+  w = np.cross(u,vp)
+
+  R = np.column_stack((u, vp, w))
+  return(R)
+
+
+class grid_featurizer:
+  def __init__(self):
+    self.protein = None
+    self.protein_graph = None 
+    self.protein_ob = None 
+    self.ligand = None
+    self.ligand_graph = None
+    self.ligand_ob = None
+    self.system = None
+    self.system_graph = None
+    self.system_ob = None
+    self.box = None
+    self.ecfp_power = 10
+    self.ecfp_degree = 2
+
+    self.box_x = 100000.0
+    self.box_y = 100000.0
+    self.box_z = 100000.0
+
+
+  def transform(self, protein_pdb, ligand_pdb, save_dir, box_x=16.0, box_y=16.0, box_z=16.0, 
+    nb_rotations = 0, nb_reflections=0, feature_types="ecfp", 
+    ecfp_degree=2, ecfp_power = 10, save_intermediates=False, ligand_only = False, **kwargs):
+    '''Takes as input files (strings) for pdb/pdbqt of the protein, pdb/pdbqt of the ligand, a filename to be saved 
+    of the merged system called system_pdb, a filename to be saved of the box called box_pdb, a filename with a .p extension
+    of the box in pickle format called box_pickle, and 3 floats for the x,y,z dimensions of the box in Angstroms
+
+    This function then computes the centroid of the ligand; decrements this centroid from the atomic coordinates of protein and
+    ligand atoms, and then merges the translated protein and ligand. This combined system/complex is then saved.
+
+    This function then removes all atoms outside of the given box dimensions, and saves the resulting box in PDB file format
+    as well as in a pickle format of the box's instance of the PDB object.
+    '''
+    a = time.time()
+    protein_name = str(protein_pdb).split("/")[len(str(protein_pdb).split("/"))-2]
+    system_pdb_file = "%s/%s.pdb" %(save_dir, protein_name)
+    system_pickle_file = "%s/%s.pickle" %(save_dir, protein_name)
+
+    features = []
+
+    self.box_x = float(box_x)/10.0
```

Comment:
Same issue here with instance variables being set in a method. In general, this is messy practice. Can you move some of this up to the constructor?